sysdeps/aero/generic/thread.cpp

```diff
@@ -17,12 +17,17 @@ extern "C" void __mlibc_enter_thread(void *entry, void *user_arg, Tcb *tcb) {
     if (mlibc::sys_tcb_set(tcb))
         __ensure(!"sys_tcb_set() failed");
 
-    void *(*func)(void *) = reinterpret_cast<void *(*)(void *)>(entry);
-    auto result = func(user_arg);
+	if(tcb->returnValueType == TcbThreadReturnValue::ThreadReturnsVoidPtr) {
+		void *(*func)(void *) = reinterpret_cast<void *(*)(void *)>(entry);
+		tcb->returnValue.voidPtr = func(user_arg);
+	} else {
+		int (*func)(void *) = reinterpret_cast<int (*)(void *)>(entry);
+		tcb->returnValue.intVal = func(user_arg);
+	}
```

Comment:
this could just be `auto`, currently you express the same thing (the type of the variable) twice. this goes for all the sysdeps